process/scToProtocol/stakingToPeer_test.go

```diff
@@ -0,0 +1,661 @@
+		return []*vmcommon.StorageUpdate{
+			{Offset: []byte("off1"), Data: []byte("data1")},
+		}, nil
+	}
+
+	peerState := &mock.AccountsStub{}
+	peerState.GetAccountWithJournalCalled = func(addressContainer state.AddressContainer) (handler state.AccountHandler, e error) {
+		peerAccount, _ := state.NewPeerAccount(&mock.AddressMock{}, &mock.AccountTrackerStub{})
+		peerAccount.Stake = big.NewInt(100)
+		peerAccount.BLSPublicKey = []byte("key")
+		return peerAccount, nil
+	}
+
+	stakingData := systemSmartContracts.StakingData{
+		StakeValue: big.NewInt(100),
+	}
+	marshalizer := &mock.MarshalizerStub{}
+	marshalizer.MarshalCalled = func(obj interface{}) (bytes []byte, e error) {
+		return json.Marshal(obj)
+	}
+	marshalizer.UnmarshalCalled = func(obj interface{}, buff []byte) error {
+		return json.Unmarshal(buff, obj)
+	}
+
+	scDataGetter := &mock.ScDataGetterMock{}
+	scDataGetter.GetCalled = func(scAddress []byte, funcName string, args ...[]byte) (bytes []byte, e error) {
+		return json.Marshal(&stakingData)
+	}
+
+	arguments := createMockArgumentsNewStakingToPeer()
+	arguments.ArgParser = argParser
+	arguments.CurrTxs = currTx
+	arguments.PeerState = peerState
+	arguments.Marshalizer = marshalizer
+	arguments.ScDataGetter = scDataGetter
+	blockBody := createBlockBody()
+	stakingToPeer, _ := NewStakingToPeer(arguments)
+
+	err := stakingToPeer.UpdateProtocol(blockBody, 0)
+	assert.Equal(t, state.ErrNilBLSPublicKey, err)
+}
+
+func TestStakingToPeer_UpdateProtocolCannotSaveAccountShouldErr(t *testing.T) {
+	t.Parallel()
+
+	testError := errors.New("error")
+	blsPublicKey := "blsPublicKey"
+	currTx := &mock.TxForCurrentBlockStub{}
+	currTx.GetTxCalled = func(txHash []byte) (handler data.TransactionHandler, e error) {
+		return &smartContractResult.SmartContractResult{
+			RcvAddr: factory.StakingSCAddress,
+		}, nil
+	}
+
+	argParser := &mock.ArgumentParserMock{}
+	argParser.GetStorageUpdatesCalled = func(data string) (updates []*vmcommon.StorageUpdate, e error) {
+		return []*vmcommon.StorageUpdate{
+			{Offset: []byte("off1"), Data: []byte("data1")},
+		}, nil
+	}
+
+	peerState := &mock.AccountsStub{}
+	peerState.GetAccountWithJournalCalled = func(addressContainer state.AddressContainer) (handler state.AccountHandler, e error) {
+		peerAccount, _ := state.NewPeerAccount(&mock.AddressMock{}, &mock.AccountTrackerStub{
+			JournalizeCalled: func(entry state.JournalEntry) {
+				return
+			},
+			SaveAccountCalled: func(accountHandler state.AccountHandler) error {
+				return testError
+			},
+		})
+		peerAccount.Stake = big.NewInt(0)
+		peerAccount.BLSPublicKey = []byte(blsPublicKey)
+		return peerAccount, nil
+	}
+
+	stakingData := systemSmartContracts.StakingData{
+		StakeValue: big.NewInt(100),
+		BlsPubKey:  []byte(blsPublicKey),
+	}
+	marshalizer := &mock.MarshalizerStub{}
+	marshalizer.MarshalCalled = func(obj interface{}) (bytes []byte, e error) {
```

Comment:
you can use the `MarshalizerMock` instead of `MarshalizerStub` in the tests where you rewrite the marshal/unmarshal logic

Reply:
I will not change this. Because in MarshalizerStub if you take a look is no implementation there.